src/main/java/persistence/sql/dml/parser/ValueParser.java

```diff
@@ -0,0 +1,51 @@
+package persistence.sql.dml.parser;
+
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import persistence.sql.dml.keygenerator.H2KeyGenerator;
+
+import java.lang.reflect.Field;
+
+public class ValueParser {
+
+    public static String insertValuesClauseParse(Field field, Object object) {
+        if (field.isAnnotationPresent(Id.class) && field.isAnnotationPresent(GeneratedValue.class)) {
+            final H2KeyGenerator h2KeyGenerator = new H2KeyGenerator();
+            return h2KeyGenerator.generator(field.getAnnotation(GeneratedValue.class).strategy());
```

Comment:
`keyGenerator` 추상화를 잘 시켜주셨네요 👍 
개인적으로 조금 아쉬운 부분이 추상화 객체를 사용하지 않고 구현체를 직접적으로 내부에서 사용하는 것이 아쉽네요 ㅜㅜ

의존성을 주입을 시킨다면 다른 DB를 사용해야될때 내부 구현코드가 덜 수정되지 않을까요?!

변경이 힘들수도 있지만 한번 고민해보셔도 좋을것 같아요~ 😄 

음... 이런느낌으로요  🤔 
```suggestion
    public static String insertValuesClauseParse(Field field, Object object, KeyGenerator keyGenerator) {
        if (field.isAnnotationPresent(Id.class) && field.isAnnotationPresent(GeneratedValue.class)) {
            return keyGenerator.generator(field.getAnnotation(GeneratedValue.class).strategy());
        }

```

Reply:
[의존 역전 원칙(Dependency Inversion Principle)](https://koseungbin.gitbook.io/wiki/books/undefined/part-2.-di/solid/dependency-inversion-principle)
고수준 모듈은 저수준 모듈의 구현에 의존해서는 안된다. 저수준 모듈이 고수준 모듈에서 정의한 추상 타입에 의존해야 한다.

Reply:
넵 반영했습니다! 
createInsertQuery() 메소드에서만 사용하는 클래스여서, 해당메소드의 인자로 추가하였습니다.